code/obj/submachine/slots.dm

```diff
@@ -6,155 +6,149 @@
 	anchored = 1
 	density = 1
 	mats = 8
+	flags = TGUI_INTERACTIVE
 	deconstruct_flags = DECON_SIMPLE
-	//var/money = 1000000
 	var/plays = 0
 	var/working = 0
 	var/obj/item/card/id/scan = null
 
 	New()
 		AddComponent(/datum/component/mechanics_holder)
-		SEND_SIGNAL(src,COMSIG_MECHCOMP_ADD_INPUT,"activate", "activateinput")
 		..()
 
-	proc/activateinput(var/datum/mechanicsMessage/inp) //make this work some day.
-		//var/list/reflist = list("ops")
-		//Topic(null,
-		return
+/* INTERFACE */
 
-	attackby(var/obj/item/I as obj, user as mob)
-		if(istype(I, /obj/item/card/id))
-			if(src.scan)
-				boutput(user, "<span class='alert'>There is a card already in the slot machine.</span>")
-			else
-				boutput(user, "<span class='notice'>You insert your ID card.</span>")
-				usr.drop_item()
-				I.set_loc(src)
-				src.scan = I
-				src.updateUsrDialog()
-		else src.attack_hand(user)
+/obj/submachine/slot_machine/ui_interact(mob/user, datum/tgui/ui)
```

Comment:
Idle thought: could we make this boilerplate into a `#define`? It's a multi-line one so maybe not, but I imagine this is going to be added a *lot* as we gradually add tgui to things.

Out of scope for this PR.

Reply:
technically yes